.github/workflows/full_stack_tests.yml

```diff
@@ -0,0 +1,514 @@
+name: Full-stack tests
+on:
+  merge_group:
+    types: [checks_requested]
+  push:
+    branches:
+      - develop
+      - release-*
+  pull_request:
+    branches:
+      - develop
+      - release-*
+
+jobs:
+  e2e_and_acceptance_coverage:
+    name: Verify all e2e/acceptance tests are included
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository so that local actions can be used
+        uses: actions/checkout@v4
+      - name: Merge develop and set up Python
+        uses: ./.github/actions/merge-develop-and-setup-python
+        with:
+          use_cache: false
+      - name: Check that all e2e and acceptance test files are captured in wdio.conf.js and core/tests/ci-test-suite-configs
+        run: make run_tests.check_tests_are_captured_in_ci
+      - name: Report failure if failed on oppia/oppia develop branch
+        if: ${{ failure() && github.event_name == 'push' && github.repository == 'oppia/oppia' && github.ref == 'refs/heads/develop'}}
+        uses: ./.github/actions/send-webhook-notification
+        with:
+          message: "The e2e/acceptance coverage test failed on the upstream develop branch."
+          webhook-url: ${{ secrets.BUILD_FAILURE_ROOM_WEBHOOK_URL }}
+  check_test_suites_to_run:
+    name: Compute which tests to run
+    runs-on: ubuntu-22.04
+    outputs:
+      TEST_SUITES_TO_RUN: ${{ steps.compute_test_suites.outputs.TEST_SUITES_TO_RUN }}
+    steps:
+      - name: Checkout repository so that local actions can be used
+        uses: actions/checkout@v4
+        with:
+          # This fetches all git refs, which is needed in the
+          # check_ci_test_suites_to_run script below.
+          fetch-depth: 0
+      - name: Merge develop and set up Python
+        uses: ./.github/actions/merge-develop-and-setup-python
+      - name: Install Oppia dependencies
+        uses: ./.github/actions/install-oppia-dependencies
+      - id: compute_test_suites
+        name: Check test suites to run
+        env:
+          # NOTE: this is temporary.
```

Comment:
NOTE -- I will drop these lines before merging the PR. The env section will go away completely.

(The "run" statement is taken from the old workflows/check_test_suites_to_run.yml.)